lib/fhir_client/ext/reference.rb

```diff
@@ -39,7 +39,7 @@ def reference_id
       end
     end
 
-    def type
+    def get_type
```

Comment:
R4 changed the Reference resource adding a `type` element, which then caused a `type` attribute to be defined on `FHIR::Reference`, that did not exist previously.  This conflicts with the `type` method in `FHIR::ReferenceExtras` which is mixed in to provide operations like `read` to `FHIR::Reference`.  I renamed the method as a work around, but I'm sure a better solution is possible.

What do you think @jawalonoski?

[STU3 Reference](https://www.hl7.org/fhir/STU3/references.html)
[R4 Reference](https://www.hl7.org/fhir/references.html)

[FHIR::STU3::Reference](https://github.com/fhir-crucible/fhir_stu3_models/blob/master/lib/fhir_stu3_models/fhir/types/Reference.rb)
[FHIR::Reference](https://github.com/fhir-crucible/fhir_models/blob/master/lib/fhir_models/fhir/types/Reference.rb)

Reply:
`get_type` isn't really idiomatic ruby. I think you should change it to something like `resource_type` instead. My 2 cents.

Reply:
Works for me!